Fresh and juicy pipeline for vercel deployment

xcfw opened the pull request:
# Description

> Vercel automated deployment that will work for production and preview environments. 

## Type of change

> Pipeline will start on each commit to the main branch and on every open pull request, creating production and preview deploys on vercel accordingly

- [ x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

> It works with free vercel account, but should be tested on team account. Please add corresponding action secrets to github before running pipeline.

# Checklist:

- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my code
- [x] I documented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

dahal commented:
This is amazing @xcfw , thanks for your contribution 🎉 

Here are few things we think would fit our expected workflow.

1. Build and deploy preview app when new PRs are created (this PR should satisfy this feature)
2. Prod deploys are not triggered when PRs are merged to `main`, instead when a release is created and published. This is to make sure we write write release notes, for each prod deploys. During build and deploy process, we grab contents from new release notes and create changelog [eg. Changelog on our site](https://envless.dev/changelog).


Here a link for reference
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

xcfw commented:
> This is amazing @xcfw , thanks for your contribution 🎉
> 
> Prod deploys are not triggered when PRs are merged to `main`...
>
> Here a link for reference https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

Thanks for a feedback, I added conditions and now the production deploy will only be triggered on `release` changed state to `published`. Also rebased with latest `main` and removed unnecessary changes to npm and yarn files from commits.

dahal commented:
This seems to be stale, closing it. Please feel free to create a new PR for just docs and www deployments to vercel. Chances are platform will be deployed elsewhere.
